File: backendApp/src/main/java/com/cinemas/dto/request/PaymentRequest.java
package com.cinemas.dto.request;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.constraints.Null;
import lombok.Data;

import java.util.List;

//@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class PaymentRequest {
    private Integer id;

    private String orderId;

    private String paymentId;

    private List<String> quantitySeat;

    private List<String> quantityDoubleSeat;

    private Float totalPrice;

    private Integer showtimeId;

    private Integer voucherId;

    private List<BookingWaterRequest> quantityWater;

    public PaymentRequest(String orderId, String paymentId, List<String> quantitySeat, List<String> quantityDoubleSeat, Float totalPrice, Integer showtimeId, Integer voucherId, List<BookingWaterRequest> quantityWater) {
        this.orderId = orderId;
        this.paymentId = paymentId;
        this.quantitySeat = quantitySeat;
        this.quantityDoubleSeat = quantityDoubleSeat;
        this.totalPrice = totalPrice;
        this.showtimeId = showtimeId;
        this.voucherId = voucherId;
        this.quantityWater = quantityWater;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getOrderId() {
        return orderId;
    }

    public void setOrderId(String orderId) {
        this.orderId = orderId;
    }

    public String getPaymentId() {
        return paymentId;
    }

    public void setPaymentId(String paymentId) {
        this.paymentId = paymentId;
    }

    public List<String> getQuantitySeat() {
        return quantitySeat;
    }

    public void setQuantitySeat(List<String> quantitySeat) {
        this.quantitySeat = quantitySeat;
    }

    public List<String> getQuantityDoubleSeat() {
        return quantityDoubleSeat;
    }

    public void setQuantityDoubleSeat(List<String> quantityDoubleSeat) {
        this.quantityDoubleSeat = quantityDoubleSeat;
    }

    public Float getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(Float totalPrice) {
        this.totalPrice = totalPrice;
    }

    public Integer getShowtimeId() {
        return showtimeId;
    }

    public void setShowtimeId(Integer showtimeId) {
        this.showtimeId = showtimeId;
    }

    public Integer getVoucherId() {
        return voucherId;
    }

    public void setVoucherId(Integer voucherId) {
        this.voucherId = voucherId;
    }

    public List<BookingWaterRequest> getQuantityWater() {
        return quantityWater;
    }

    public void setQuantityWater(List<BookingWaterRequest> quantityWater) {
        this.quantityWater = quantityWater;
    }
}

File: fllmbooking_app/lib/data/models/PaymentRequest.dart
import 'BookingWaterRequest.dart';

class PaymentRequest {
  int? id;
  String? orderId;
  String? paymentId;
  List<String>? quantitySeat;
  List<String>? quantityDoubleSeat;
  double? totalPrice;
  int? showtimeId;
  int? voucherId;
  List<BookingWaterRequest>? quantityWater;

  PaymentRequest({
    this.id,
    this.orderId,
    this.paymentId,
    this.quantitySeat,
    this.quantityDoubleSeat,
    this.totalPrice,
    this.showtimeId,
    this.voucherId,
    this.quantityWater,
  });

  // Convert to JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'orderId': orderId,
      'paymentId': paymentId,
      'quantitySeat': quantitySeat ?? [],
      'quantityDoubleSeat': quantityDoubleSeat ?? [],
      'totalPrice': totalPrice,
      'showtimeId': showtimeId,
      'voucherId': voucherId,
      'quantityWater': quantityWater?.map((e) => e.toJson()).toList(),
    };
  }

  // Create from JSON
  factory PaymentRequest.fromJson(Map<String, dynamic> json) {
    return PaymentRequest(
      id: json['id'],
      orderId: json['orderId'],
      paymentId: json['paymentId'],
      quantitySeat: (json['quantitySeat'] as List?)?.map((item) => item as String).toList() ?? [],
      quantityDoubleSeat: (json['quantityDoubleSeat'] as List?)?.map((item) => item as String).toList() ?? [],
      totalPrice: json['totalPrice'],
      showtimeId: json['showtimeId'],
      voucherId: json['voucherId'],
      quantityWater: (json['quantityWater'] as List?)
          ?.map((item) => BookingWaterRequest.fromJson(item))
          .toList(),
    );
  }

  @override
  String toString() {
    return 'PaymentRequest{id: $id, orderId: $orderId, paymentId: $paymentId, quantitySeat: $quantitySeat, quantityDoubleSeat: $quantityDoubleSeat, totalPrice: $totalPrice, showtimeId: $showtimeId, voucherId: $voucherId, quantityWater: $quantityWater}';
  }
}
